src/edu/cscc/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,96 @@
+package edu.cscc;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+public class CustomerRepositoryImpl implements CustomerRepository {
+
+    private Set<Customer> customers;
+
+    public CustomerRepositoryImpl(){
+        customers = new HashSet<>();
+    }
+
+    public Customer create(Customer customer) throws DuplicateCustomerException {
+        boolean foundMatch = customers.stream()
+                .anyMatch(customer1 -> customer1.getEmailAddress().equals(customer.getEmailAddress()));
+
+        if (foundMatch) {
+            throw new DuplicateCustomerException(customer.getEmailAddress());
+        } else {
+            customer.setAccountNumber(UUID.randomUUID());
+            customers.add(customer);
+            return customer;
+        }
+    }
+
+    public Customer read(UUID accountNumber){
+        Optional<Customer> accountFound = customers.stream()
+                .filter(customer -> customer.getAccountNumber() == accountNumber)
+                .findFirst();
+
+        if (accountFound.isPresent()) {
+            return accountFound.get();
+        } else {
+            return null;
+        }
```

Comment:
This is totally valid, but could be simplified to:
```suggestion
        return accountFound.get();
```